https://issues.redhat.com/browse/ACM-7118 (#5762)
open-cluster-management/rhacm-docs: clusters/cluster_lifecycle/adv_config_cluster.adoc

*Note:* Managed clusters must use the host name `api.<cluster_name>.<base_domain>` to access the hub cluster. You cannot use named certificates that are configured with other host names.

[#configuring-HTTP-or-HTTPS-proxy-between-hub-cluster-and-managed-cluster]

> == Configuring HTTP or HTTPS proxy between hub cluster and managed cluster

Reviewer: I recommend that we shorten the anchor and update the title 
```suggestion
[#configuring-proxy-hub-and-managed-cluster]
== Configuring the proxy between hub cluster and managed cluster
```

> To register a managed cluster to your Multicluster Engine (MCE) hub cluster, you need to transport the  managed cluster to your MCE hub cluster. Sometimes your managed cluster cannot directly reach your MCE hub cluster. In this instance, configure the proxy settings to allow the communications from the managed cluster to access the MCE hub cluster through a HTTP or HTTPS proxy server

Reviewer: Remember to use the conrefs for product names, which can be found in the [common-attributes.adoc file](https://github.com/stolostron/rhacm-docs/blob/2.9_stage/modules/common-attributes.adoc)
```suggestion
To register a managed cluster to your {mce-short} hub cluster, you need to transport the  managed cluster to your {mce-short} hub cluster. Sometimes your managed cluster cannot directly reach your {mce-short} hub cluster. In this instance, configure the proxy settings to allow the communications from the managed cluster to access the {mce-short} hub cluster through a HTTP or HTTPS proxy server.
```

> For example, the MCE hub cluster is in a public cloud, and the managed cluster is in a private cloud environment behind firewalls. The communications out of the private cloud can only go through a HTTP or HTTPS proxy server. 

Reviewer: use the conref for multicluster engine operator

> === Prerequisites

Reviewer: Throughout the doc, Prerequisites are Header 2 (H2). There are a couple of file that have Prerequisites at H3, and I would like to check on that post GA
```suggestion
== Prerequisites
```

> - You have a HTTP or HTTPS proxy server running that supports HTTP tunnels. For example, HTTP connect method. 

Reviewer: ```suggestion

- You have a HTTP or HTTPS proxy server running that supports HTTP tunnels. For example, HTTP connect method. 
```

> - You have a HTTP or HTTPS proxy server that can reach the managed clusters and can access the MCE hub cluster. 

Reviewer 2: The managed cluster can reach the proxy server and the proxy server can access the hub cluster.

> - You have a manged cluster that can reach the HTTP or HTTPS proxy server, and the proxy server can access the MCE hub cluster. 

Reviewer: use the conref here for multicluster engine operator

> . Create a `KlusterConfig` with proxy settings. 

Reviewer: ```suggestion
. Create a `KlusterConfig` resource with proxy settings. 
```

.. See the following configuration with HTTP proxy:
+
[source,yaml]
----
apiVersion: config.open-cluster-management.io/v1alpha1
kind: KlusterletConfig
metadata:
  name: http-proxy
spec:
  hubKubeAPIServerProxyConfig:
    httpProxy: "http://<username>:<password>@<ip>:<port>"
----
+ 
.. See the following configuration with HTTPS proxy:  
+
[source,yaml]
----
apiVersion: config.open-cluster-management.io/v1alpha1
kind: KlusterletConfig
metadata:
  name: https-proxy
spec:
  hubKubeAPIServerProxyConfig:
    httpsProxy: "https://<username>:<password>@<ip>:<port>"
    caBundle: <user-ca-bundle> 
----
+ 

> *Note:* A CA certificate is required if the HTTPS proxy server is severing with a self-signed server certificate. The HTTPS proxy is used if both HTTP proxy and HTTPS proxy are specified.

Reviewer 2: > A CA certificate is required if the HTTPS proxy server is severing with a self-signed server certificate.

Change the above to 
> A CA certificate is required when the HTTPS proxy server is configured.

> . When creating a managed cluster, chose the `KlusterletConfig` by adding an annotation that refers to the `KlusterletConfig`. See the following example:

Reviewer: ```suggestion

. When creating a managed cluster, choose the `KlusterletConfig` resource by adding an annotation that refers to the `KlusterletConfig` resource. See the following example:
```

> . When creating a managed cluster, chose the `KlusterletConfig` by adding an annotation and referring to the `ManagedCluster`. See the following example:

Reviewer 2: The annotation refers to `KlusterletConfig` instead of `ManagedCluster`

+
[source,yaml]
----
apiVersion: cluster.open-cluster-management.io/v1
kind: ManagedCluster
metadata:
  annotations:
    agent.open-cluster-management.io/klusterlet-config: <klusterlet-config-name>
  name:<managed-cluster-name>
spec:
  hubAcceptsClient: true
  leaseDurationSeconds: 60 
----
+ 

> *Note:* You might need to toggle the yaml view to add the annotation to the `ManagedCluster` when you operate on the MCE UI console.

Reviewer: ```suggestion
*Note:* You might need to toggle the YAML view to add the annotation to the `ManagedCluster` resource when you operate on the {mce-short} console.
```

> [#disabling-HTTP-or-HTTPS-proxy-between-hub-cluster-and-managed-cluster]
> == Disabling HTTP or HTTPS proxy between hub cluster and managed cluster

Reviewer: Similar to an earlier suggestion, shorten the anchor and update the title
```suggestion
[#disabling-proxy-between-hub-and-managed-cluster]
== Disabling the proxy between hub cluster and managed cluster
```

> If your development needs change, you might need to disable the HTTP or HTTPS proxy. 

Reviewer: needs to change 

OR 

If your development changes, you might...

> . Go to `ManagedCluster`.

Reviewer: Where can the resource be accessed? Is there a tab that the user needs to select? 
```suggestion
. Go to `ManagedCluster` resource.
```

*Note:* Managed clusters must use the host name `api.<cluster_name>.<base_domain>` to access the hub cluster. You cannot use named certificates that are configured with other host names.

[#configuring-proxy-hub-and-managed-cluster]
== Configuring the proxy between hub cluster and managed cluster

To register a managed cluster to your {mce-short} hub cluster, you need to transport the  managed cluster to your {mce-short} hub cluster. Sometimes your managed cluster cannot directly reach your {mce-short} hub cluster. In this instance, configure the proxy settings to allow the communications from the managed cluster to access the {mce-short} hub cluster through a HTTP or HTTPS proxy server.

For example, the {mce-short} hub cluster is in a public cloud, and the managed cluster is in a private cloud environment behind firewalls. The communications out of the private cloud can only go through a HTTP or HTTPS proxy server. 

== Prerequisites

- You have a HTTP or HTTPS proxy server running that supports HTTP tunnels. For example, HTTP connect method. 
- You have a manged cluster that can reach the HTTP or HTTPS proxy server, and the proxy server can access the {mce-short} hub cluster. 

> 

Reviewer: ```suggestion

Complete the following steps to configure the proxy settings between hub cluster and managed cluster:

```
